common/persistence/sql/sql_execution_store.go

```diff
@@ -375,6 +375,21 @@ func (m *sqlExecutionStore) GetWorkflowExecution(
 			Message: fmt.Sprintf("GetWorkflowExecution: failed. Error: %v", err),
 		}
 	}
+	// if we have checksum, we need to make sure the rangeID did not change
+	// if the rangeID changed, it means the shard ownership might have changed
+	// and the workflow might have been updated when we read the data, so the data
+	// we read might not be from a consistent view, the checksum validation might fail
+	// in that case, we clear the checksum data so that we will not perform the validation
```

Comment:
let's mention "The GetWorkflowExecution operation will not be impacted by this. `ChecksumData` is purely for validation purposes."